src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -925,4 +925,23 @@ public function testGetEnvCastsNull($expected, string $prefix)
             });
         }));
     }
+
+    /**
+     * @dataProvider provideGetEnvDefined
+     */
+    public function testGetEnvDefined(bool $expected, callable $callback)
+    {
+        $this->assertSame($expected, (new EnvVarProcessor(new Container()))->getEnv('defined', 'NO_SOMETHING', $callback));
+    }
+
+    public static function provideGetEnvDefined(): iterable
+    {
+        $returnValue = fn ($value) => fn () => $value;
+
+        yield 'Defined' => [true, $returnValue('foo')];
```

Comment:
I don't think `$returnValue` is necessary. The other way is less verbose.
```suggestion
        yield 'Defined' => [true, fn () => 'foo'];
```

Reply:
Fair enough